packages/app/src/data-sync/services/data-sync.service.ts

```diff
@@ -3,81 +3,68 @@
-import DeviceBackupAdapter from "Backend/adapters/device-backup/device-backup-adapter.class"
-import DeviceService from "Backend/device-service"
-import { DataSyncClass } from "App/data-sync/services/data-sync-class.interface"
-import path from "path"
-import getAppPath from "App/main/utils/get-app-path"
-import { DeviceResponseStatus } from "Backend/adapters/device-response.interface"
-import { SyncFileSystemService } from "App/data-sync/services/sync-file-system.service"
-import { InitializeOptions } from "App/data-sync/types"
-import { SyncFileSystemServiceFactory } from "App/data-sync/services/sync-file-system-service-factory"
 
-const syncCatalogName = "sync"
-const cacheCatalogName = "cache"
-const cacheFileNamse: Record<DataIndex, string> = {
-  [DataIndex.Contact]: "contacts.json",
-  [DataIndex.Message]: "messages.json",
-  [DataIndex.Thread]: "threads.json",
-}
-
-export class DataSync implements DataSyncClass {
-  private token = ""
-  private serialNumber = ""
+export class DataSyncService {
   private contactIndexer: ContactIndexer | null = null
   private messageIndexer: MessageIndexer | null = null
   private threadIndexer: ThreadIndexer | null = null
-  private syncFileSystemService: SyncFileSystemService | null = null
-  public indexesMap: Map<DataIndex, SerialisedIndexData<any>> = new Map()
+  // TODO implement device backup service and use it instead of adapter
+  private deviceBackupService: DeviceBackup
 
   constructor(
+    private index: IndexStorage,
     private deviceService: DeviceService,
-    private deviceBackup: DeviceBackupAdapter
-  ) {}
-
-  async initialize({
-    token,
-    serialNumber,
-  }: InitializeOptions): Promise<boolean> {
-    this.token = token
-    this.serialNumber = serialNumber
-    this.initializeAllIndexers(token)
-
-    return await this.loadsIndexesFromCache()
-  }
+    private keyStorage: MetadataStore,
+    private fileSystemStorage: FileSystemService
+  ) {
+    this.deviceBackupService = new DeviceBackup(
+      new DeviceBaseInfo(this.deviceService),
+      new DeviceBackupService(this.deviceService),
+      new DeviceFileSystem(this.deviceService)
+    )
 
-  initializeAllIndexers(token: string): void {
-    this.syncFileSystemService = SyncFileSystemServiceFactory.create(token)
     this.contactIndexer = new ContactIndexer(
-      this.syncFileSystemService,
+      this.fileSystemStorage,
       new ContactPresenter()
     )
     this.messageIndexer = new MessageIndexer(
-      this.syncFileSystemService,
+      this.fileSystemStorage,
       new MessagePresenter()
     )
     this.threadIndexer = new ThreadIndexer(
-      this.syncFileSystemService,
+      this.fileSystemStorage,
       new ThreadPresenter()
     )
   }
 
-  async indexAll(): Promise<void> {
-    if (this.deviceBackup.backuping) {
-      return
-    }
+  public async indexAll(): Promise<void> {
+    const serialNumber = this.keyStorage.getValue(
+      MetadataKey.DeviceSerialNumber
+    ) as string
```

Comment:
casting is safer over the assertion
```suggestion
    const serialNumber = String(this.keyStorage.getValue(
      MetadataKey.DeviceSerialNumber
    ))
```